lib/ClimaCoreTempestRemap/src/netcdf.jl

```diff
@@ -0,0 +1,213 @@
+        Spaces.FiniteDifferenceSpace{S}(space.vertical_topology),
+    )
+end
+
+"""
+    spacedims(space::Spaces.AbstractSpace)
+
+The names of the NetCDF dimensions used by `space`.
+"""
+spacedims(space::Spaces.SpectralElementSpace2D) = ("ncol",)
+spacedims(space::Spaces.CenterFiniteDifferenceSpace) = ("z",)
+spacedims(space::Spaces.CenterExtrudedFiniteDifferenceSpace) = ("ncol", "z")
+
+"""
+    define_var!(nc::NCDataset, name::String, T, space::AbstractSpace, extradims=())
+
+Define a new variable in `nc` named `name` of element type `T` suitable for
+storing a field on `space`, along with any further dimensions specified in
+`extradims`.
+"""
+function define_var!(
+    nc::NCDataset,
+    name::String,
+    T,
+    space::Spaces.AbstractSpace,
+    extradims = (),
+)
+    defVar(nc, name, Float64, (spacedims(space)..., extradims...))
+end
+
+
+function add_time_coordinate!(
+    nc::NCDataset,
+    len = Inf;
+    units = "seconds since 2020-01-01 00:00:00",
+    kwargs...,
+)
+    defDim(nc, "time", len)
+    time = defVar(nc, "time", Float64, ("time",))
+    time.attrib["long_name"] = "time"
+    time.attrib["axis"] = "T"
+    time.attrib["units"] = units
+    for (k, v) in kwargs
+        time.attrib[String(k)] = v
+    end
+    return time
+end
+
+"""
+    write_var_cgll!(nc::NCDataset, name::AbstractString, field::Fields.AbstractField; sliceidx...)
+
+Write `field` to the NetCDF dataset `nc` as a variable named `name`.
+
+If the variable does not already exist, it will be added.
+
+Optional keyword arguments allow writing the field to a slice of a variable
+along extra dimensions. The name specifies the dimension, and the value is the
+index in that dimension (the dimension must already exist).
+
+# Example
+```
+write_var_cgll!(nc, "u", u, time=1)
+```
+"""
+function write_var_cgll!(
+    nc::NCDataset,
+    name::AbstractString,
+    field::Fields.Field;
+    sliceidx...,
+)
+    space = axes(field)
+    slicedims = string.(keys(sliceidx))
+    if !haskey(nc, name)
+        var = define_var!(nc, name, eltype(Float64), space, slicedims)
+    else
+        var = nc[name]
+    end
+    reqdims = [spacedims(space)..., slicedims...]
+    # check the dimensions are correct
+    if collect(NCDatasets.dimnames(var)) != reqdims
+        error(
+            "NetCDF variable $name has incorrect dimensions: $(NCDatasets.dimnames(var)), required $(reqdims)",
+        )
+    end
+    write_cgll_data!(var, field, space, values(sliceidx)...)
+end
+
+"""
+    write_cgll_data!(var, field::Fields.AbstractField, space::Spaces.AbstractSpace, extraidx...)
+
+Write the data in `field` (on `space`) to `var`. `extraidx` are any extra indices of `var`.
+"""
+function write_cgll_data!(
+    var,
+    field,
+    space::Spaces.SpectralElementSpace2D,
+    extraidx...,
+)
+    nodes = Spaces.unique_nodes(space)
+    data = Fields.field_values(field)
+    for (col, ((i, j), e)) in enumerate(nodes)
+        var[col, extraidx...] = slab(data, e)[i, j]
```

Comment:
do we need to check if these are scalars?

Reply:
I've added an assert